### PR TITLE
feat(client): Migrate <Link> to ts

### PR DESCRIPTION
**Pull request by thecodingaviator**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

#42256

### CODE REVIEW

**Comment by thecodingaviator** [comment]
<Link> is used in multiple other components. I'm not sure of a better way to approach this but from what I understand, this PR will need to be merged in order for those to be migrated. One example of a component is client/src/components/Footer .

cc: @raisedadead

File: client/src/components/helpers/declarations.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line import/unambiguous
+declare module 'react-test-renderer';
```
Comment (ojeytonwilliams) [inline client/src/components/helpers/declarations.d.ts:2]: Could you import `@types/react-test-renderer` instead?

I should have put this in the original issue, but I'm still figuring this stuff out.  Sorry!

Comment (thecodingaviator) [reply client/src/components/helpers/declarations.d.ts:2]: No worries. You want me to import it in place of line 2 above. Right?

Comment (ojeytonwilliams) [reply client/src/components/helpers/declarations.d.ts:2]: Sorry, I shouldn't have said 'import', my bad.  As I understand it, you just need to `npm -D @types/react-test-renderer` inside the `client` directory and TypeScript will look for it automatically.

**Review by raisedadead — COMMENTED** [review]
Hey there,

Thank you for taking the time to contribute to the TypeScript migration. 

The main issue has been updated a lot over the last few days, and so we suggest you re-read it and update this pull request to match if you have the time.

The main goals for a PR are:
- [x] Changing the filename from `PascalCase` to `kebab-case` and ensuring that the imports in code are all still valid.
- [x] Changing the file extension to .tsx (if a file has React JSX syntax)
- [x] Fixing as many TypeScript and Eslint errors as you can while refactoring the code.

If you need any assistance, please reach out in our [contributing chat room](https://chat.freecodecamp.org/channel/contributors).

**Comment by thecodingaviator** [comment]
Hi @raisedadead I have gone ahead and checked the boxes off. I believe the PR should be fine now.

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1:

**Review by naomi-lgbt — DISMISSED** [review]

**Comment by ojeytonwilliams** [comment]
I'll fix these conflicts

**Pull request merged**